src/averageAges.js

```diff
@@ -14,12 +14,32 @@
  *
  * @return {number}
  */
+const SEX_MALE = 'm';
+const SEX_FEMALE = 'f';
+const CENTURY_VALUE = 100;
+
+const getAverageAge = (acc, person, personIndex, currentArray) => {
+  const { born, died } = person;
+  const age = died - born;
+
+  if (personIndex === currentArray.length - 1) {
+    return (acc + age) / currentArray.length;
+  }
+
+  return acc + age;
+};
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  return people
+    .filter(({ sex, died }) => {
+      const deathCentury = Math.ceil(died / CENTURY_VALUE);
+      const isCenturyRequired = century
+        ? sex === SEX_MALE && deathCentury === century
+        : sex === SEX_MALE;
+
+      return isCenturyRequired;
+    })
+    .reduce(getAverageAge, 0);
```

Comment:
It is better to completely transfer the calculation logic to another function and transfer only the array that needs to be calculated.
```js
const filteredMen = people.filter(...);
return getAverageAge(filteredMen);
...
const getAverageAge = filteredPeople => {
  return filteredPeople.reduce(...);
};
```
The same for `calculateWomenAverageAge` function.